### PR TITLE
Documented the ErrorRenderer component

### LINKED ISSUE #11837: Add docs for ErrorCatcher component
| Q            | A
| ------------ | ---
| Feature PR   | symfony/symfony#31065
| PR author(s) | @yceruto
| Merged in | 4.4

### PR DESCRIPTION
**Pull request by javiereguiluz**
This fixes #11837.

I'll need help for some things:

* I don't understand well how this component is enabled/disabled/configured both as a stand-alone component and inside a Symfony application.
* I'm not sure if I removed the right things from Debug component. Should we make more changes?

Thanks.

### CODE REVIEW

File: components/error_catcher.rst [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+
+Rendering PHP Errors and Exceptions
+-----------------------------------
+
+Another feature provided by this component are the "error renderers", which
+converts PHP errors and exceptions into other formats such as JSON and HTML::
+
+    use Symfony\Component\ErrorHandler\ErrorRenderer\ErrorRenderer;
+    use Symfony\Component\ErrorHandler\ErrorRenderer\HtmlErrorRenderer;
+    use Symfony\Component\ErrorHandler\ErrorRenderer\JsonErrorRenderer;
```
Comment (yceruto) [inline components/error_catcher.rst]: `\ErrorCatcher\`, same everywhere

File: components/error_catcher.rst [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+
+.. tip::
+
+    If the ``getformat()`` method of your error renderer matches one of formats
+    supported by the built-in renderers, the built-in renderer is replaced by
+    your custom renderer.
+
+To enable the new error renderer in the application,
+:ref:`register it as a service <service-container-creating-service>` and
+:doc:`tag it </service_container/tags>` with the ``error_handler.renderer``
```
Comment (yceruto) [inline components/error_catcher.rst]: `error_catcher.renderer`, same bellow

File: components/error_catcher.rst [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+.. include:: /components/require_autoload.rst.inc
+
+Usage
+-----
+
+The ErrorCatcher component provides several handlers and renderers to convert
+PHP errors and exceptions into other formats easier to debug when working with
+HTTP applications.
+
+.. TODO: how are these handlers enabled in the app? (Previously: Debug::enable())
```
Comment (yceruto) [inline components/error_catcher.rst]: Yes, they are enabled by `Debug::enable()`, but we're still debating the correct namespace in https://github.com/symfony/symfony/pull/32303

**Review by yceruto — COMMENTED** [review]
Just a quick review.

Thank you for caring about this!

File: components/error_catcher.rst [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+    use Symfony\Component\ErrorHandler\ErrorRenderer\ErrorRenderer;
+    use Symfony\Component\ErrorHandler\ErrorRenderer\HtmlErrorRenderer;
+    use Symfony\Component\ErrorHandler\ErrorRenderer\JsonErrorRenderer;
+
+    $renderers = [
+        new HtmlErrorRenderer(),
+        new JsonErrorRenderer(),
+        // ...
+    ];
+    $errorRenderer = new ErrorRenderer($renderers);
```
Comment (yceruto) [inline components/error_catcher.rst]: `new ErrorFormatter`

File: components/error_catcher.rst [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+                $content['@context']['exceptions'] = $exception->toArray();
+            }
+
+            return (string) json_encode($content);
+        }
+    }
+
+.. tip::
+
+    If the ``getformat()`` method of your error renderer matches one of formats
```
Comment (ihmels) [inline components/error_catcher.rst]: Must be `getFormat()`. 😊

**Comment by yceruto** [comment]
This document needs to be divided, now the ErrorRenderer component contains only the renderer layer and the new ErrorHandler component (https://github.com/symfony/symfony/pull/32471) contains the handlers.

**Comment by OskarStark** [comment]
@yceruto is this still necessary or can we close this PR?

**Comment by yceruto** [comment]
I think the ErrorRenderer part is still good. I will try to update it according to the latest changes.

**Comment by yceruto** [comment]
See https://github.com/symfony/symfony-docs/pull/12187, Javier, feel free to continue your work there.

**Comment by javiereguiluz** [comment]
Let's close this in favor of #12187.

**Pull request closed without merging**